client/Speedy/GameScene.swift

```diff
@@ -213,60 +211,35 @@ class GameScene : SKScene, SKPhysicsContactDelegate {
         if touchedNode is SKScene {
             // can't move the scene, finger probably fell off a circle?
             return
+        }else if touchedNode.parent! is NumberCircle{
+            //Only number circles can be moved
+            touchedNode.position.x = touchLocation.x
+            touchedNode.position.y = touchLocation.y
         }
-        touchedNode.position.x = touchLocation.x
-        touchedNode.position.y = touchLocation.y
-        
-        addPoint(touchLocation)
     }
     
     override func touchesEnded(touches: NSSet, withEvent event: UIEvent) {
-        wayPoints.removeAll(keepCapacity: false)
+        let touch = touches.anyObject() as UITouch
+        let touchLocation = touch.locationInNode(self)
+        var touchedNode = nodeAtPoint(touchLocation)
         
+        if touchedNode is SKLabelNode {
+            touchedNode = touchedNode.parent!.parent!
+            
+        }
+        
+        //Break the node joint if touch is released
+        breakJoint()
```

Comment:
just want to make sure, if a valid joint is made from number to operator to number, the current joint will be nil, and this function won't break a joint?